Specs/Scene/Cesium3DTilesetSpec.js

```diff
@@ -0,0 +1,633 @@
+        ];
+
+        return when.all(promises, function(tilesets) {
+            // Root tiles are ready, now zoom in to request child tiles
+            viewAllTiles();
+            scene.renderForSpecs();
+
+            // Maximum of 6 requests allowed. Expect the first tileset to use four,
+            // and the second tileset to use the remaining two
+            expect(tilesets[0]._statistics.numberOfPendingRequests).toEqual(4);
+            expect(tilesets[1]._statistics.numberOfPendingRequests).toEqual(2);
+        });
+    });
+
+    it('load progress events are raised', function() {
+        // [numberOfPendingRequests, numberProcessing]
+        var results = [
+            [1, 0],
+            [0, 1],
+            [0, 0]
+        ];
+        var i = 0;
+
+        viewNothing();
+        return Cesium3DTilesTester.loadTileset(scene, tilesetUrl).then(function(tileset) {
+            tileset.loadProgress.addEventListener(function(numberOfPendingRequests, numberProcessing) {
+                var expected = results[i++];
+                expect(numberOfPendingRequests).toEqual(expected[0]);
+                expect(numberProcessing).toEqual(expected[1]);
+            });
+            viewRootOnly();
+            scene.renderForSpecs();
+            return Cesium3DTilesTester.waitForPendingRequests(scene, tileset);
+        });
+    });
+
+    it('tile visible event is raised', function() {
+        viewRootOnly();
+        return Cesium3DTilesTester.loadTileset(scene, tilesetUrl).then(function(tileset) {
+            tileset.tileVisible.addEventListener(function(tile) {
+                expect(tile).toBe(tileset._root);
+                expect(tileset._root.visibility(scene.frameState.cullingVolume)).not.toEqual(CullingVolume.MASK_OUTSIDE);
+            });
+
+            scene.renderForSpecs();
+        });
+    });
+
+    it('destroys', function() {
+        return Cesium3DTilesTester.loadTileset(scene, tilesetUrl).then(function(tileset) {
+            var root = tileset._root;
+            expect(tileset.isDestroyed()).toEqual(false);
+            scene.primitives.remove(tileset);
+            expect(tileset.isDestroyed()).toEqual(true);
+
+            // Check that all tiles are destroyed
+            expect(root.isDestroyed()).toEqual(true);
+            expect(root.children[0].isDestroyed()).toEqual(true);
+            expect(root.children[1].isDestroyed()).toEqual(true);
+            expect(root.children[2].isDestroyed()).toEqual(true);
+            expect(root.children[3].isDestroyed()).toEqual(true);
+        });
+    });
+
+    it('destroys before loadTilesJson finishes', function() {
+        var tileset = scene.primitives.add(new Cesium3DTileset({
+            url : tilesetUrl
+        }));
+        scene.primitives.remove(tileset);
+        // TODO : hard to test this one (check that promise never resolves or rejects?)
+        // Should destroy function reject ready promise?
```

Comment:
The behavior is up to you, but it should be consistent with `Model` (changes to `Model` are OK).